gtimelog: unstable-2023-10-05 -> 0.12.0

oxzi opened the pull request:
## Description of changes

Version bump from a commit to the latest release.

- https://github.com/gtimelog/gtimelog/releases/tag/0.12.0
- https://github.com/gtimelog/gtimelog/blob/0.12.0/CHANGES.rst#0120-2024-04-03

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Sigmanificient commented on pkgs/by-name/gt/gtimelog/package.nix:
```diff
@@ -4,14 +4,14 @@
     repo = pname;
-    rev = "ba606cbe8eef0e3dc098c6ab3bcbe381bf7ef410";
-    hash = "sha256-+iBHfbUJtAtI/vcHj0Y8f9OxAp1SnhQyMqedVzSYPZQ=";
+    rev = version;
```
```suggestion
    rev = "refs/tags/${version}";
```

Sigmanificient commented on pkgs/by-name/gt/gtimelog/package.nix:
```diff
@@ -4,14 +4,14 @@
 
   src = fetchFromGitHub {
     owner = pname;
     repo = pname;
```
- https://github.com/NixOS/nixpkgs/issues/277994

SuperSandro2000 commented on pkgs/by-name/gt/gtimelog/package.nix:
```diff
@@ -4,14 +4,14 @@
   pname = "gtimelog";
-  version = "unstable-2023-10-05";
+  version = "0.12.0";
   format = "setuptools";
```
While we are at it, can we change this to pyproject = true?

oxzi replied:
Did the change, please take another look.

oxzi committed:
gtimelog: unstable-2023-10-05 -> 0.12.0

- https://github.com/gtimelog/gtimelog/releases/tag/0.12.0
- https://github.com/gtimelog/gtimelog/blob/0.12.0/CHANGES.rst#0120-2024-04-03

oxzi commented:
@pbsds: Thank you very much for picking up this PR and merging it :)

pbsds commented:
If a reviewer goes unresponsive it is fine bump the PR and even request review from someone else. ;)
